src/transformers/trainer.py

```diff
@@ -2476,27 +2495,46 @@ def floating_point_ops(self, inputs: Dict[str, Union[torch.Tensor, Any]]):
 
     def init_git_repo(self):
         """
-        Initializes a git repo in :obj:`self.args.push_to_hub_model_id`.
+        Initializes a git repo in :obj:`self.args.hub_model_id`.
         """
-        if not self.args.should_save:
+        if not self.is_world_process_zero():
             return
         use_auth_token = True if self.args.hub_token is None else self.args.hub_token
         if self.args.hub_model_id is None:
             repo_name = get_full_repo_name(Path(self.args.output_dir).name, token=self.args.hub_token)
         else:
             repo_name = self.args.hub_model_id
 
-        self.repo = Repository(
-            self.args.output_dir,
-            clone_from=repo_name,
-            use_auth_token=use_auth_token,
-        )
+        try:
+            self.repo = Repository(
+                self.args.output_dir,
+                clone_from=repo_name,
+                use_auth_token=use_auth_token,
+            )
+        except EnvironmentError:
+            if self.args.overwrite_output_dir:
+                # Try again after wiping output_dir
+                shutil.rmtree(self.args.output_dir)
+                self.repo = Repository(
+                    self.args.output_dir,
+                    clone_from=repo_name,
+                    use_auth_token=use_auth_token,
+                )
```

Comment:
In which case would it error out here?

Reply:
Is it safe to remove the `output_dir` rather than raise the error?

Reply:
This errors out if the output dir exists and the repo is not a local clone of the remote model id passed. Since `--overwrite_output_dir` is passed, I think it's safe to wipe it down in this case.

Reply:
Sounds good